eng/Publishing.props

```diff
@@ -3,28 +3,30 @@
   <Import Project="$(RepositoryEngineeringDir)VmrLayout.props" />
 
   <PropertyGroup>
+    <PublishingVersion>4</PublishingVersion>
     <EnableDefaultArtifacts>false</EnableDefaultArtifacts>
-    <PushToLocalStorage>true</PushToLocalStorage>
     <PreserveRepoOrigin>true</PreserveRepoOrigin>
-    <ArtifactsStagingDir Condition="'$(ArtifactsStagingDir)' == ''">$(ArtifactsDir)/staging/</ArtifactsStagingDir>
-    <SourceBuiltAssetsDir>$(ArtifactsStagingDir)/assets/$(Configuration)</SourceBuiltAssetsDir>
-    <SourceBuiltAssetManifestsDir>$(ArtifactsStagingDir)/manifests/$(Configuration)</SourceBuiltAssetManifestsDir>
-    <SourceBuiltPdbArtifactsDir>$(ArtifactsStagingDir)/PDB/$(Configuration)</SourceBuiltPdbArtifactsDir>
     <PublishingUseHardlinksIfPossible>true</PublishingUseHardlinksIfPossible>
     <AssetManifestFileName Condition="'$(MergedAssetManifestName)' != ''">$(MergedAssetManifestName)</AssetManifestFileName>
     <!-- Do not auto-generate symbol packages for assets. These are already generated by individual repo builds. -->
     <AutoGenerateSymbolPackages>false</AutoGenerateSymbolPackages>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' != 'true'">
-    <SourceBuiltShippingPackagesDir>$(ArtifactsStagingDir)/packages/$(Configuration)/Shipping</SourceBuiltShippingPackagesDir>
-    <SourceBuiltNonShippingPackagesDir>$(ArtifactsStagingDir)/packages/$(Configuration)/NonShipping</SourceBuiltNonShippingPackagesDir>
-  </PropertyGroup>
   <!--
     When building from source, the Private.SourceBuilt.Artifacts archive will contain the nuget packages, so don't push them where we'll upload from.
+    As we place the packages in a different directory structure than the staging directory,
+    we can't use PublishingVersion 4 as the paths to the packages won't be a subdirectory of the artifacts (which is fine as we don't publish assets for source-only builds).
```

Comment:
From what I understand, the eventual goal is to get everyone on V4 publishing. Given that we would need to solve this for SB as well. Please file a tracking issue for removing this.

Reply:
I'll file the tracking issue along with the PR into the main branch.

Reply:
ArtifactsStagingDir is under a separate directory from ArtifactsDir when running on Azdo.

I can change to using something like "ArtifactsStagingTmpDir" so it will live in a subdirectory.

Reply:
FutureArtifactPublishBasePath is initialized too late for me to be able to easily intercept.

Let's leave fixing this to main branch PR.